SignalServiceKit/src/Storage/OWSStorage.m

```diff
@@ -76,7 +76,10 @@ - (void)readWriteWithBlock:(void (^)(YapDatabaseReadWriteTransaction *transactio
     OWSAssert(delegate);
     OWSAssert(delegate.areAllRegistrationsComplete || self.canWriteBeforeStorageReady);
 
-    OWSBackgroundTask *backgroundTask = [OWSBackgroundTask backgroundTaskWithLabelStr:__PRETTY_FUNCTION__];
+    OWSBackgroundTask *_Nullable backgroundTask = nil;
+    if (CurrentAppContext().isMainApp) {
```

Comment:
Background tasks have no effect in SAE and will only clutter the logs with useless warnings.